src/Hadrian/Haskell/Cabal.hs

```diff
@@ -15,30 +15,32 @@ module Hadrian.Haskell.Cabal (
 
 import Development.Shake
 
-import Hadrian.Haskell.Cabal.Parse
+import Context.Type
+import Hadrian.Haskell.Cabal.Type       as C
+import Hadrian.Haskell.Cabal.Configured as CC
 import Hadrian.Package
 import Hadrian.Oracles.TextFile
 
 -- | Read a Cabal file and return the package version. The Cabal file is tracked.
-pkgVersion :: FilePath -> Action String
-pkgVersion cabalFile = version <$> readCabalFile cabalFile
+pkgVersion :: Context -> Action (Maybe String)
+pkgVersion = fmap (fmap C.version) . readCabalFile
 
 -- | Read a Cabal file and return the package identifier, e.g. @base-4.10.0.0@.
 -- The Cabal file is tracked.
-pkgIdentifier :: FilePath -> Action String
-pkgIdentifier cabalFile = do
-    cabal <- readCabalFile cabalFile
-    return $ if null (version cabal)
-        then name cabal
-        else name cabal ++ "-" ++ version cabal
+pkgIdentifier :: Context -> Action String
+pkgIdentifier ctx = do
+    Just cabal <- readCabalFile ctx
```

Comment:
Looks like this function is partial: please note this in the comment and print an informative error message.

Reply:
Fixed in an upcoming commit.